forc-pkg/src/manifest.rs

```diff
@@ -61,6 +69,38 @@ impl ManifestFile {
             .parent()
             .expect("failed to retrieve manifest directory")
     }
+
+    /// Returns manifest file map from member name to the corresponding package manifest file
+    pub fn member_manifests(&self) -> Result<MemberManifestFiles> {
+        let mut member_manifest_files = BTreeMap::new();
+        match self {
+            ManifestFile::Package(pkg_manifest_file) => {
+                let member_name = &pkg_manifest_file.project.name;
+                member_manifest_files.insert(member_name.clone(), *pkg_manifest_file.clone());
```

Comment:
Thinking on this more, even if we have a `ManifestFile::Package`, if we're within a workspace we still want to produce *all* members. We should use our new `PackageManifestFile::workspace` method (mentioned in my previous comment) to check if the package is within a workspace first, and if so, produce the manifest for all members (as long as our package is one of the members). This is to ensure the lock file (and in turn build plan) remains consistent for the whole workspace.

In the follow-up PR when it comes time to `build`/`check`/`test`/`deploy` a single package, we can skip the nodes in the `compilation_order` that aren't a part of the single package's dependencies to avoid building the entire workspace every time.